### ISSUE TITLE
skeleton attributes and typescript support

### ISSUE THREAD

**Issue by atoy40** [grommet/grommet#6771]
Hi,

the skeleton property, for example for a Text component, accept any parameters (according to the documentation) of the component itself to be override when rendererd in skeleton "mode".
But the typescript type for the skeleton parameter is very restricted and limited to this SkeletonType :
```typescript
export type SkeletonType =
  | boolean
  | {
      animation?: AnimateType;
      colors?: SkeletonColorsType;
      depth?: number;
      message?: string | { start?: string; end?: string };
    };
```

Is there any workaround ? disabling ts checking ? or any other ? or is it planned to be improved ? or do I miss something :D ?

Thanks

Anthony.

**Comment by MikeKingdom** [comment]
@atoy40 Good catch and thanks for the issue. It looks like the only values supported in `Text` for the skeleton prop should be what's supported for `Skeleton` (e.g. `SkeletonProps`) plus `size`. I don't see a workaround other than disabling type checking for a usage of this property.

**Comment by tdonhauser** [comment]
I have a similar problem. When I create a Box like this:
```
<Box
  skeleton={{
    animation: "fadeIn",
  }}
/>
```
I get this error:
`Type error: Type 'string' is not assignable to type 'boolean | undefined'.`
Could this be related to this?
